node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/AllocatableClusterResources.java

```diff
@@ -182,6 +180,7 @@ private static NodeResources averageRealResourcesOf(List<Node> nodes, NodeReposi
 
                 if ( ! between(applicationLimits.min().nodeResources(), applicationLimits.max().nodeResources(), advertisedResources)) continue;
                 if ( ! systemLimits.isWithinRealLimits(realResources, clusterSpec.type())) continue;
```

Comment:
I originally thought so as well, but a PR in internal repo failed without this check since candidates with wrong architecture was chosen (test seems to has copied most of the code in `from()` to test class). But I might be missing something...

Reply:
I think that must be something else but we can merge this and see if you want.

Reply:
Looks to me as `between()` will not verify compatibility if neither min nor max is specified

Reply:
Yes, but that means no node resources, and therefore no architecture is specified.

Reply:
You are right, this is not necessary, closing PR